### PR TITLE
Add complex math, Mandelbrot node, minor fixes

### PR DESCRIPTION
**Pull request by luisbarrancos**
Add complex numbers arithmetic functions, and box, sphere folding
functions for Mandelbox as well as colouring functions for the
Mandelbrot (2D) node.

### CODE REVIEW

File: src/appleseed.shaders/include/appleseed/math/as_math_complex.h [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef AS_COMPLEX_HELPERS_H
+#define AS_COMPLEX_HELPERS_H
+
+struct Complex {
```
Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/math/as_math_complex.h]: Move brace to its own line.

File: src/appleseed.shaders/include/appleseed/math/as_math_complex.h [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+};
+
+//
+// Reference:
+//
+//      Complex Number Arithmetic
+//      https://en.wikipedia.org/wiki/Complex_number
+//
+
+void add_complex(Complex augend, Complex addend, output Complex sum)
```
Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/math/as_math_complex.h]: augend? Never heard this term :)

In appleseed we usually call these arguments lhs and rhs, for left-hand side and right-hand side.

File: src/appleseed.shaders/include/appleseed/math/as_math_complex.h [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef AS_COMPLEX_HELPERS_H
```
Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/math/as_math_complex.h]: Should be `AS_MATH_COMPLEX_H`.

File: src/appleseed.shaders/include/appleseed/math/as_math_complex.h (hunk 2 above)
Comment (luisbarrancos) [inline src/appleseed.shaders/include/appleseed/math/as_math_complex.h]: @dictoon if one wants to be strict about naming :) , we could've used addends for both, but this way everything just documents itself in an unambiguous way, though it's slightly more verbose.
[arithmetic operators: formal naming convention](https://math.stackexchange.com/questions/975541/what-are-the-formal-names-of-operands-and-results-for-basic-operations)

Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/math/as_math_complex.h]: Interesting, thanks :) These terms are rather unusual, but lets keep them for the sake of enlightment :)

**Pull request merged**